### DISCUSSION TITLE
Directing auth to signIn versus signUp

### DISCUSSION THREAD

**Discussion by fadethegap** [supabase/supabase#5407] [Questions] [answered]
Using third party providers is no problem. The auth signIn function works great.

But considering email signUp/signIn, what is the best technique to direct email subscribers to the correct auth (signUp/signIn) call?

![Screen Shot 2022-02-09 at 11 41 21 PM](https://user-images.githubusercontent.com/47059192/153338800-2fa7da9c-ed85-40c4-b355-674cb0692630.png)
![Screen Shot 2022-02-09 at 11 41 41 PM](https://user-images.githubusercontent.com/47059192/153338822-292df7c9-80cb-4292-93c2-0fdb69b79664.png)
![Screen Shot 2022-02-09 at 11 41 56 PM](https://user-images.githubusercontent.com/47059192/153338823-f9dc45e3-5ea9-4c7c-915b-a9225c8325b1.png)

@GaryAustin1

**Comment by GaryAustin1** [comment]
Can you offer more info on what you are running into, or what you want to do?  I don't use react anymore, seems like you might be asking about process idea and not actual code issue.  Not clear.

**Reply by fadethegap** [reply]
Thank you for your response Gary, I REALLY appreciate your time and effort!!!

The challenge was directing the email login form to the correct auth function. When using third party auth it is straightforward, signIn is always used.

However, when using an email login without Magic Link, then they need to be routed to signUp if they do NOT have an account. And signIn otherwise.

I was trying to find a supabase method to handle this. In the end I created an api route to check whether the user exists, and routed appropriately from there.

**Comment by fadethegap** [comment, marked as answer]
If someone has a more elegant solution please share it with me. But I resolved the issue by creating an api route and checking to see if the user exists. Then routing to the appropriate function (signUp/signIn).

Api route:
![Screen Shot 2022-02-11 at 9 20 25 AM](https://user-images.githubusercontent.com/47059192/153608182-b11432be-776c-4b5b-82a1-bc81b8c15050.png)

Handle routing to appropriate function:
![Screen Shot 2022-02-11 at 9 14 39 AM](https://user-images.githubusercontent.com/47059192/153608293-9533dfe5-57dc-41b4-89ae-3c65bc1cf85f.png)
